src/pages/chart/chart.tsx

```diff
@@ -62,32 +64,29 @@ const Chart = observer(({ show_digits_stats }: { show_digits_stats: boolean }) =
     useEffect(() => {
         return () => {
             chart_api.api.forgetAll('ticks');
-            Object.keys(subscriptions).forEach(subscription_id => {
-                requestForgetStream(subscription_id);
-            });
         };
     }, []);
 
+    useEffect(() => {
+        chartSubscriptionIdRef.current = chart_store.chartSubscriptionId;
+    }, [chart_store.chartSubscriptionId]);
+
     useEffect(() => {
         if (!symbol) updateSymbol();
     }, [symbol, updateSymbol]);
 
     const requestAPI = (req: ServerTimeRequest | ActiveSymbolsRequest | TradingTimesRequest) => {
         return chart_api.api.send(req);
     };
-
-    const requestForget = () => {
-        return chart_api.api.forgetAll('ticks');
-    };
-
     const requestForgetStream = (subscription_id: string) => {
-        subscriptions[subscription_id as string]?.unsubscribe();
-        delete subscriptions[subscription_id];
+        subscription_id && chart_api.api.forget(subscription_id);
     };
 
     const requestSubscribe = async (req: TicksStreamRequest, callback: (data: any) => void) => {
         try {
+            requestForgetStream(chartSubscriptionIdRef.current);
             const history = await chart_api.api.send(req);
+            setChartSubscriptionId(history?.subscription.id);
             if (history) callback(history);
             if (req.subscribe === 1) {
                 subscriptions[history?.subscription.id] = chart_api.api
```

Comment:
@mayuran-deriv do we need this line now ? Since we are storing the chart subscription on every render.
`if (req.subscribe === 1) {
subscriptions[history?.subscription.id] = chart_api.api }`

Reply:
hey @rupato-deriv 
we are sending this call back no 
so i kept that 
  subscriptions[history?.subscription.id] = chart_api.api
                    .onMessage()
                    ?.subscribe(({ data }: { data: TicksHistoryResponse }) => {
                        **callback(data);**
                    });